plugin/nfs/nfs.py

```diff
@@ -0,0 +1,421 @@
+        """Remove an entry by its export id"""
+        output = []
+        for exp in exports:
+            eid = NFSPlugin._export_id(exp[0], exp[1])
+            if export_id != eid:
+                output.append(exp)
+        return output
+
+    @staticmethod
+    def _write_exports(exports, filename=_EXPORTS):
+        """write out the exports list to a file"""
+        try:
+            efile = open(filename, "w")
+        except OSError:
+            raise LsmError(ErrorNumber.PLUGIN_BUG,
+                           'error opening exports for writing:')
+        try:
+            efile.write('# NFS exports managed by libstoragemgmt.'
+                        ' do not edit.\n')
+            for exp in exports:
+                efile.write("%s %s(%s)\n" % exp)
+        except IOError:
+            raise LsmError(ErrorNumber.PLUGIN_BUG,
+                           'error writing exports')
+        finally:
+            efile.close()
+
+    @staticmethod
+    def _add_export(path, host="*", opts=None):
+        """Add an export to the list"""
+        if opts is None:
+            opts = []
+
+        try:
+            exports = NFSPlugin._load_exports()
+            exports = NFSPlugin._filter_export(exports,
+                                               NFSPlugin._export_id(path, host))
+            exports.append((path, host, ",".join(opts)))
+            NFSPlugin._write_exports(exports)
+        except Exception:
+            raise
+
+    @staticmethod
+    def _update_exports():
+        """Update the system"""
+        cmd = ["/usr/sbin/exportfs", "-ar"]
+        try:
+            subprocess.check_call(cmd)
+        except subprocess.CalledProcessError:
+            raise LsmError(ErrorNumber.INVALID_ARGUMENT,
+                           'exportfs failed')
+        except OSError:
+            raise LsmError(ErrorNumber.PLUGIN_BUG,
+                           'error calling exportfs')
+
+    def plugin_info(self, flags=0):
+        return "Local NFS Exports", VERSION
+
+    def __init__(self):
+        self.tmo = 0
+        return
+
+    def plugin_register(self, uri, password, timeout, flags=0):
+        self.tmo = timeout
+        if os.geteuid() != 0:
+            raise LsmError(ErrorNumber.INVALID_ARGUMENT,
+                           "This plugin requires root privilege for both " \
+                           "daemon and client")
+
+        return
+
+    def time_out_set(self, ms, flags=0):
+        self.tmo = ms
+
+    def time_out_get(self, flags=0):
+        return self.tmo
+
+    def pools(self, search_key=None, search_value=None, flags=0):
+        pools = []
+        parts = list_mounts()
+        for p in parts:
+            try:
+                st = os.statvfs(p)
+                total_size = st.f_blocks * st.f_frsize
+                avail_size = st.f_bavail * st.f_frsize
+                fsid = get_fsid(p)
+                pooltype =  Pool.ELEMENT_TYPE_VOLUME | Pool.ELEMENT_TYPE_VOLUME_FULL
```

Comment:
Currently, the pool abstracted by nfs plugin does not hold any volume. Try `Pool.ELEMENT_TYPE_FS`.